### PR TITLE
My solution

### PR DESCRIPTION
**Pull request by kovaliskoveronika**
car_no_user
car_with_user
create_driver
drivers_list
update_license

### CODE REVIEW

File: templates/taxi/driver_confirm_delete.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% extends "base.html" %}
+
+{% block content %}
+  <h1>Delete driver <i>{{ driver.username }}</i>?</h1>
+  <form action="" method="post">
+    {% csrf_token %}
+
+    <input type="submit" value="Yes" class="btn btn-danger">
+  </form>
+{% endblock %}
```
Comment (Oleg-s-s) [inline templates/taxi/driver_confirm_delete.html]: blank line

File: templates/taxi/driver_form.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+{% block content %}
+  <h1>Create driver</h1>
+  <form action="" method="post" novalidate>
+    {% csrf_token %}
+    {{ form|crispy }}
+
+    <input type="submit" value="Submit" class="btn btn-primary">
+  </form>
+{% endblock %}
```
Comment (Oleg-s-s) [inline templates/taxi/driver_form.html]: blank line

File: templates/taxi/license-form.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+{% block content %}
+  <h1>Update license</h1>
+  <form action="" method="post" novalidate>
+    {% csrf_token %}
+    {{ form|crispy }}
+
+    <input type="submit" value="Submit" class="btn btn-primary">
+  </form>
+{% endblock %}
```
Comment (Oleg-s-s) [inline templates/taxi/license-form.html]: same

**Review by Oleg-s-s — CHANGES_REQUESTED** [review]

File: taxi/urls.py [hunk 4]
```diff
@@ -46,6 +51,31 @@
...
+    ),
+    path(
+        "drivers/car/<int:car_pk>/<int:driver_pk>/update/",
+        driver_car_update,
+        name="assign-car"
+    ),
+    path(
+        "drivers/car/<int:car_pk>/<int:driver_pk>/delete/",
+        driver_car_delete,
+        name="reassign-car"
```
Comment (YuriiKindrat) [inline taxi/urls.py]: why do u need 2 pk here? U can use only one here
also, it seems to me, that url should start from car/car_pk/f.e. update driver 
cause u are using this views from car-detail page

File: taxi/views.py [hunk 5]
```diff
@@ -87,3 +89,35 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
...
+    driver.save()
+    return redirect("taxi:car-detail", pk=car_pk)
+
+
+def driver_car_delete(request, car_pk, driver_pk):
+    car = get_object_or_404(Car, id=car_pk)
+    driver = get_object_or_404(Driver, id=driver_pk)
+    driver.cars.remove(car)
+    driver.save()
+    return redirect("taxi:car-detail", pk=car_pk)
```
Comment (YuriiKindrat) [inline taxi/views.py]: make ur code DRY here, u can use only one view here

File: templates/taxi/driver_confirm_delete.html (hunk 1 above)
Comment (YuriiKindrat) [reply templates/taxi/driver_confirm_delete.html]: didn't fix that

File: templates/taxi/driver_form.html (hunk 2 above)
Comment (YuriiKindrat) [reply templates/taxi/driver_form.html]: +

**Review by YuriiKindrat — CHANGES_REQUESTED** [review]

File: templates/taxi/car_detail.html [hunk 6]
```diff
@@ -12,7 +12,22 @@ <h1>
...
+        {% csrf_token %}
+        <input type="hidden" name="action" value="remove">
+        <button type="submit" class="btn btn-danger link-to-page">Delete me from this car</button>
+      </form>
+      {% else %}
+        <form method="post" action="{% url 'taxi:driver-car-update' car_pk=car.pk %}">
+          {% csrf_token %}
+          <input type="hidden" name="action" value="add">
+          <button type="submit" class="btn btn-primary link-to-page">Assign me to this car</button>
+        </form>
```
Comment (Oleg-s-s) [inline templates/taxi/car_detail.html]: You can call {% if %} in form after csrf token to avoid unnecessary code repeating

**Review by Oleg-s-s — CHANGES_REQUESTED** [review]

**Review by Oleg-s-s — APPROVED** [review]
Good job

**Pull request closed without merging**